pennylane/beta/tapes/qubit_param_shift.py

```diff
@@ -0,0 +1,220 @@
+    consider the expectation value of an observable :math:`O`:
+
+    .. math:: f(p_i)  = \langle O \rangle(p_i) = \langle 0 | U(p_i)^\dagger O U(p_i) | 0\rangle.
+
+    The gradient of this expectation value can be calculated using :math:`2N` evaluations
+    using the parameter-shift rule:
+
+    .. math:: \frac{\partial f}{\partial p_i} = \frac{1}{2\sin s} \left[ f(p_i + s) - f(p_i -s) \right].
+
+    **Gradients of variances**
+
+    We can extend this to the variance, :math:`g(p_i)=\langle O^2 \rangle (p_i) - \langle O \rangle(p_i)^2`,
+    by noting that:
+
+    .. math::
+
+        \frac{\partial g}{\partial p_i}= \frac{\partial}{\partial p_i} \langle O^2 \rangle (p_i)
+        - 2 f(p_i) \frac{\partial f}{\partial p_i}.
+
+    This results in :math:`4N + 1` evaluations.
+
+    In the case where :math:`O` is involutory (:math:`O^2 = I`), the first term in the above
+    expression vanishes, and we are simply left with
+
+    .. math:: \frac{\partial g}{\partial p_i} = - 2 f(p_i) \frac{\partial f}{\partial p_i},
+
+    allowing us to compute the gradient using :math:`2N + 1` evaluations.
+    """
+
+    def _update_circuit_info(self):
+        super()._update_circuit_info()
+
+        # set parameter_shift as the analytic_pd method
+        self.analytic_pd = self.parameter_shift
+
+        # check if the quantum tape contains any measurements
+        self.var_mask = [m.return_type is qml.operation.Variance for m in self.measurements]
+
+        if any(self.var_mask):
+            # The tape contains variances.
+            # Set parameter_shift_var as the analytic_pd method
+            self.analytic_pd = self.parameter_shift_var
+
+            # Make a copy of the original measurements; we will be mutating them
+            # during the parameter shift method.
+            self._original_measurements = self._measurements.copy()
+
+            # Finally, store the locations of any variance measurements in the
+            # measurement queue.
+            self.var_idx = np.where(self.var_mask)[0]
+
+    def _grad_method(self, idx, use_graph=True, default_method="F"):
+        op = self._par_info[idx]["op"]
+
+        if op.grad_method == "F":
+            return "F"
+
+        return super()._grad_method(idx, use_graph=use_graph, default_method="A")
+
+    def jacobian(self, device, params=None, **options):
+        # The parameter_shift_var method needs to evaluate the circuit
+        # at the unshifted parameter values; these are stored in the
+        # self._evA attribute. Here, we set the value of the attribute to None
+        # before each Jacobian call, so that the expectation value is calculated only once.
+        self._evA = None
+        return super().jacobian(device, params, **options)
+
+    def parameter_shift(self, idx, device, params, **options):
+        r"""Partial derivative using the parameter-shift rule of a tape consisting of *only*
+        expectation values of observables.

```

Comment:
Is the point here to select specifically the `expval()` return type, or just to exclude the `var()` return type? I'm thinking because it sounds like `probs()` is excluded (even though they also are technically expectation values)

Reply:
Nope, this _just_ excludes variances. I'll update.

Reply:
![image](https://user-images.githubusercontent.com/2959003/93660704-c2e14800-fa90-11ea-95ff-a01586387439.png)